sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2714,6 +2726,17 @@ class Dataset[T] private[sql](
    */
   def withColumn(colName: String, col: Column): DataFrame = withColumns(Seq(colName), Seq(col))
 
+  /**
+   * Returns a new Dataset by selecting a metadata column with the given logical name.
+   *
+   * A metadata column can be accessed this way even if the underlying data source defines a data
+   * column with a conflicting name.
+   *
+   * @group untypedrel
+   * @since 4.0.0
+   */
+  def withMetadataColumn(colName: String): DataFrame = withColumn(colName, metadataColumn(colName))
```

Comment:
people can just do `df.withColumn(name, df.metadataColumn(col))`, right?

Reply:
This is a weird API as there is no `withColumn` function that takes only a string parameter.

Reply:
Removed `withMetadataColumn` method for now. Nothing stops us from adding it in the future if it's needed.